packages/ui/src/components/shared/settings/__test__/ChangeWalletPasswordDialog.spec.tsx

```diff
@@ -0,0 +1,131 @@
+import { UserEvent } from '@testing-library/user-event/setup/setup';
+import { initializeKeyring, newUser, render, screen, waitFor } from '__tests__/testUtils';
+import { SettingsDialogScreen } from 'types';
+import SettingsWalletButton from '../SettingsWalletButton';
+
+describe('ChangeWalletPasswordDialog', () => {
+  let user: UserEvent;
+  beforeEach(async () => {
+    user = newUser();
+
+    render(<SettingsWalletButton />, {
+      preloadedState: {
+        app: { seedReady: true, ready: true, locked: false },
+        settingsDialog: { settingsDialogScreen: SettingsDialogScreen.ChangeWalletPassword },
+      },
+    });
+
+    const settingsButton = screen.getByTitle('Open settings');
+    await user.click(settingsButton);
+  });
+
+  describe('VerifyingPassword', () => {
+    it('should show content of ChangeWalletPasswordDialog', async () => {
+      await waitFor(() => {
+        expect(screen.getByText(/Change wallet password/)).toBeInTheDocument();
+        expect(screen.getByText(/Enter your wallet password to continue/)).toBeInTheDocument();
+        expect(screen.getByRole('button', { name: /Continue/ })).toBeDisabled();
+        expect(screen.getByRole('button', { name: /Back/ })).toBeInTheDocument();
+      });
+    });
+
+    it('should enable `Continue` button when typing password', async () => {
+      const passwordField = await screen.findByLabelText(/Your wallet password/);
+      await user.type(passwordField, 'testing-password');
+
+      expect(await screen.findByRole('button', { name: /Continue/ })).toBeEnabled();
+    });
+
+    it('should show error message on submitting incorrect password', async () => {
+      await initializeKeyring();
+
+      const passwordField = await screen.findByLabelText(/Your wallet password/);
+      await user.type(passwordField, 'incorrect-password');
+
+      const continueButton = await screen.findByRole('button', { name: /Continue/ });
+      await user.click(continueButton);
+
+      expect(await screen.findByText(/Password incorrect/)).toBeInTheDocument();
+    });
+
+    it('should switch to default settings dialog content when clicking on `Back` button', async () => {
+      const backButton = await screen.findByRole('button', { name: /Back/ });
+      await user.click(backButton);
+
+      await waitFor(() => {
+        expect(screen.getByRole('button', { name: /Dark/ })).toBeInTheDocument();
+        expect(screen.getByRole('button', { name: /English/ })).toBeInTheDocument();
+        expect(screen.getByRole('button', { name: /5 minutes/ })).toBeInTheDocument();
+        expect(screen.getByRole('button', { name: /Backup secret recovery phrase/ })).toBeInTheDocument();
+        expect(screen.getByRole('button', { name: /Change wallet password/ })).toBeInTheDocument();
```

Comment:
This's probably good for now but we should think of a better way to check if we're in the setting screen other than checking all the buttons like we have it here, since we can't always remember to add another `expect` into this list when adding a new setting in the future, not to mention we have 4 different places to update.